salt/modules/zcbuildout.py

```diff
@@ -186,23 +186,20 @@ def by_level(self):
 LOG = _Logger()
 
 
-def _encode_string(string):
-    if isinstance(string, six.text_type):
-        string = string.encode('utf-8')
-    return string
-
-
 def _encode_status(status):
-    status['out'] = _encode_string(status['out'])
-    status['outlog_by_level'] = _encode_string(status['outlog_by_level'])
+    if status['out'] is None:
+        status['out'] = None
```

Comment:
what is if this doing?

Reply:
@rares-pop Yeah, that's a fair question. Basically my thought was to just be explicit that `None` was a valid/desired `status['out']`.

My other option of course was to flip the logic and just have: `if status['out'] is not None: <do encoding>`, and then just implicitly leave `status['out']` alone. 

Which would certainly work, just like this does.

The biggest problem for me is just one of priority. I wouldn't even have been in this code in the first place if the tests weren't using `/bin/false`, which was breaking tests elsewhere. I went to fix that problem throughout the codebase and discovered that these tests were just not even turned on. So I figured I'd see if I could improve that a little, without spending *too* much time on it.

Combine that with the fact that this doesn't appear to be a widely (at all?) used module, and the tests have been completely disabled and definitely broken, for quite some time, and according to the modules own docstring it's only a beta support. I also found it kind of difficult to tell at a glance the *why* behind most of this module. This was one of those things. Why isn't it just doing `stderr.decode()`? It would take more time than I had to devote.

Of course, if someone *does* have the time and interest, I would 💯% support their efforts to improve this entire module, and be more than happy to review their PRs 👍

Reply:
@waynew - Hehe.. a comment would do it too. :) I have nothing against your change, I was just curious about the reasons behind. Thanks for the reply!